src/Compilers/Core/MSBuildTask/BuildClient.cs

```diff
@@ -32,13 +32,15 @@ internal static class BuildClient
         // Spend up to 20s connecting to a new process, to allow time for it to start.
         private const int TimeOutMsNewProcess = 20000;
 
+        private static bool IsRunningOnWindows => Path.DirectorySeparatorChar == '\\';
```

Comment:
Do we actually care if it is windows or if it is anything where backslash is a primary separator?

IsRunningOnWindows still seems the shortest way to call this